content/research/market-health/posts/2026-05-11-okx/analysis.py

```
@@ -0,0 +1,114 @@
+    after = ""
+    while len(candles) < limit:
+        params = {"instId": inst_id, "bar": "1H", "limit": 100}
+        if after:
+            params["after"] = after
+        r = requests.get("https://www.okx.com/api/v5/market/history-candles", params=params, timeout=15)
+        r.raise_for_status()
+        batch = r.json()["data"]
+        if not batch:
+            break
+        for c in batch:
+            ts, o, h, l, close, vol = c[0], c[1], c[2], c[3], c[4], c[5]
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)